frontend/app/components/api/api-v3/hal-resources/work-package-resource.service.ts

```diff
@@ -67,6 +67,29 @@ function wpResource(
       return !this[fieldName] && fieldSchema.writable && fieldSchema.required;
     }
 
+    public allowedValuesFor(field):op.HalResource[] {
+      var deferred = $q.defer();
```

Comment:
Instead of using the `$q` service, you could return the promise returned by `this.getForm()`.
To get the values, you resolve at the moment, you would just have to use a return statement instead of `deferred.resolve`.